### PR TITLE
[fix bug] Execute upgrade in a different path

### PR DESCRIPTION
**Pull request by Shuimo03**
issues: https://github.com/devstream-io/devstream/issues/1172

## Pre-Checklist

Note: please complete **_ALL_** items in the following checklist.

- [x] I have read through the [CONTRIBUTING.md](https://github.com/devstream-io/devstream/blob/main/CONTRIBUTING.md) documentation.
- [x] My code has the necessary comments and documentation (if needed).
- [x] I have added relevant tests

## Description

Binary executable file path( /usr/local/bin/):
avatar

exec dtm upgrade other path(/root/devstreamTest)
avatar

## Related Issues

https://github.com/devstream-io/devstream/issues/1172

### CODE REVIEW

**Comment by aFlyBird0** [comment]
Hi @Shuimo03 , thanks for your contribution! Please [join CNCF's Slack](https://slack.cncf.io), and join our [channel in CNCF Slack](https://cloud-native.slack.com/messages/devstream)!
Here are a few useful things for you to continue your open-source journey:
- If you are searching for something else to work on, please see ["find an issue"](https://docs.devstream.io/en/latest/contributing_guide/#find-an-issue).
- Check out our [new contributor guide](https://docs.devstream.io/en/latest/contributing_guide/#new-contributor-guide) and the ["Developer Guide"](https://docs.devstream.io/en/latest/development/development-workflow/).
- Read our [blog](https://blog.devstream.io/); they are on [Medium](https://medium.com/devstream), too.


A kind reminder that you haven't signed off on your commit(s) yet. Check out our [official doc](https://docs.devstream.io/en/latest/contributing_guide/#sign-your-commits) to learn how to do this. Also, DevStream community follows the [conventional commits specification](https://www.conventionalcommits.org/eXn/v1.0.0/). Please read our [doc](https://docs.devstream.io/en/latest/development/commit-messages/) and update your commit message.

File: internal/pkg/upgrade/upgrade.go [hunk 1]
```diff
@@ -20,6 +21,7 @@ const (
 	dtmBakFileName = "dtm-bak"
 	dtmOrg         = "devstream-io"
 	dtmRepo        = "devstream"
+	dtmExecute     = "dtm"
```
Comment (daniel-hutao) [inline internal/pkg/upgrade/upgrade.go:24]: The filename with DevStream cli file may not be "dtm". Users could rename it to any.

Here's some hint for you:

```go
binFilePath, err := os.Executable()
dtmFileDir, dtmFileName = filepath.Split(binFilePath)
```

**Review by daniel-hutao — CHANGES_REQUESTED** [review]

File: internal/pkg/upgrade/upgrade.go (hunk 1 above)
Comment (aFlyBird0) [reply internal/pkg/upgrade/upgrade.go:24]: This has been resolved in #1133 .

@Shuimo03 Please use this variable:

https://github.com/devstream-io/devstream/blob/30cf3afdf2bea912217ff33a3b770a0dea6d1b72/internal/pkg/upgrade/upgrade.go#L25-L26

**Review by aFlyBird0 — CHANGES_REQUESTED** [review]

File: internal/pkg/upgrade/upgrade.go (hunk 1 above)
Comment (Shuimo03) [reply internal/pkg/upgrade/upgrade.go:24]: thank you, I try variable

File: internal/pkg/upgrade/upgrade.go [hunk 2]
```diff
@@ -47,6 +43,13 @@ func Upgrade(continueDirectly bool) error {
 
 	log.Debugf("Dtm upgrade: dtm file name is : %v", dtmFileName)
 
+	execDir, err := os.Executable()
```
Comment (aFlyBird0) [inline internal/pkg/upgrade/upgrade.go:46]: Why not just use this func?😂

image

Comment (Shuimo03) [reply internal/pkg/upgrade/upgrade.go]: I just fixed this error

**Comment by aFlyBird0** [comment]
## Tested:

### same path(`./`)
image

### different path
image

image

Maybe we could remind users to use `sudo upgrade` if "open /dtm-tmp.tmp" failed.(it means `dtm` binary is in the path where the current user can not access to). What's your opinion? @Shuimo03

**Comment by IronCore864** [comment]
@Shuimo03 hi, how's it going? Needs a small update on the commit message lint. Don't worry though, take your time.

Also, a kind reminder, you can link this PR to the issue. See [here](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) about how to do this.

**Comment by IronCore864** [comment]
@Shuimo03, a gentle reminder: this PR has been stale for a week.
When you have time, please review the comments above and try to merge this one. Thanks :)

**Review by aFlyBird0 — APPROVED** [review]

**Pull request merged**